### ISSUE TITLE
feature request: range search for datasets

### ISSUE THREAD

**Issue by andy-esch** [CartoDB/bigmetadata#369]
Searching the Data Observatory for measures in specific years is tricky since the timerange metadata is a string. What would be much more useful is if there was a `daterange` property, where we could do ask questions like "what are all the measures for tags x in the year 2013?".

With `daterange`s, we could easily encapsulate the time part like so:
```sql
...
WHERE timerange @> to_date('2013'::text, 'YYYY')
```

Reducing it to this we don't need to know a priori all the timerange tags (e.g., `2010 - 2014`, `2013q4`, etc.)

cc @hannahblue

**Comment by hannahblue** [comment]
I agree this change will be helpful. @juanignaciosl do you foresee any issues with this?

**Comment by juanignaciosl** [comment]
Yep, it might not be easy. This is related to #351 (as a "project"). @antoniocarlon and @javitonino are considering different implementations for the timespan improvements, maybe this can be addressed as well. What do you think, folks?

**Comment by antoniocarlon** [comment]
Yes, now that we are going to modify the timespans to make them full citizens we can take this into account
